src/code/z_skelanime.c

```diff
@@ -1,680 +1,684 @@
+            POLY_OPA_DISP = &polyTemp[2];
             mtx++;
         } else {
-            if (dList[0] != NULL) {
-                SysMatrix_GetStateAsRSPMatrix(mtx);
+            if (limbDList != NULL) {
+                Matrix_ToMtx(mtx);
                 mtx++;
             }
         }
     }
 
     if (postLimbDraw != NULL) {
-        postLimbDraw(globalCtx, 1, &dList[0], &rot, actor);
+        postLimbDraw(globalCtx, 1, &limbDList, &rot, arg);
     }
 
-    if (limbEntry->firstChildIndex != LIMB_DONE) {
-        SkelAnime_DrawLimbSV(globalCtx, limbEntry->firstChildIndex, skeleton, limbDrawTable, overrideLimbDraw,
-                             postLimbDraw, actor, &mtx);
+    if (rootLimb->child != LIMB_DONE) {
+        SkelAnime_DrawFlexLimbOpa(globalCtx, rootLimb->child, skeleton, jointTable, overrideLimbDraw, postLimbDraw, arg,
+                                  &mtx);
     }
 
-    SysMatrix_StatePop();
+    Matrix_StatePop();
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
 }
 
-void func_80134148(GlobalContext* globalCtx, s32 limbIndex, void** skeleton, Vec3s* limbDrawTable,
-                   OverrideLimbDraw overrideLimbDraw, PostLimbDraw postLimbDraw, UnkActorDraw unkDraw, Actor* actor,
-                   RSPMatrix** mtx) {
-    StandardLimb* limbEntry;
-    Gfx* dList[2];
+void func_80134148(GlobalContext* globalCtx, s32 limbIndex, void** skeleton, Vec3s* jointTable,
+                   OverrideLimbDrawOpa overrideLimbDraw, PostLimbDrawOpa postLimbDraw, UnkActorDraw unkDraw, void* arg,
+                   Mtx** mtx) {
+    StandardLimb* limb;
+    Gfx* newDList;
+    Gfx* limbDList;
     Vec3f pos;
     Vec3s rot;
-    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
-    s32 pad2;
 
-    SysMatrix_StatePush();
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    Matrix_StatePush();
 
-    limbEntry = (StandardLimb*)Lib_SegmentedToVirtual(skeleton[limbIndex]);
+    limb = (StandardLimb*)Lib_SegmentedToVirtual(skeleton[limbIndex]);
     limbIndex++;
 
-    rot = limbDrawTable[limbIndex];
-    pos.x = limbEntry->translation.x;
-    pos.y = limbEntry->translation.y;
-    pos.z = limbEntry->translation.z;
+    rot = jointTable[limbIndex];
+    pos.x = limb->jointPos.x;
+    pos.y = limb->jointPos.y;
+    pos.z = limb->jointPos.z;
 
-    dList[0] = limbEntry->displayLists[0];
-    dList[1] = dList[0];
+    newDList = limbDList = limb->dList;
 
-    if ((overrideLimbDraw == NULL) || (overrideLimbDraw(globalCtx, limbIndex, &dList[1], &pos, &rot, actor) == 0)) {
-        SysMatrix_RotateAndTranslateState(&pos, &rot);
-        SysMatrix_StatePush();
-        unkDraw(globalCtx, limbIndex, actor);
-        if (dList[1] != NULL) {
-            Gfx* polyTemp = gfxCtx->polyOpa.p;
+    if ((overrideLimbDraw == NULL) || !overrideLimbDraw(globalCtx, limbIndex, &newDList, &pos, &rot, arg)) {
+        Matrix_JointPosition(&pos, &rot);
+        Matrix_StatePush();
+        unkDraw(globalCtx, limbIndex, arg);
```

Comment:
```suggestion
        //! @bug Does not check unkDraw is not NULL before calling it.
        unkDraw(globalCtx, limbIndex, arg);
```
And elsewhere too in this file.